calico_node/startup/startup.go

```diff
@@ -303,11 +316,21 @@ func configureIPsAndSubnets(node *api.Node) {
 		}
 	} else {
 		if ipv6Env != "" {
-			node.Spec.BGP.IPv6Address = parseIPEnvironment("IP6", ipv6Env, 6)
+			envIp6 := parseIPEnvironment("IP6", ipv6Env, 6)
+			node.Spec.BGP.IPv6Address = envIp6
 		}
 		validateIP(node.Spec.BGP.IPv6Address)
 	}
 
+	// Detect if we've seen the IP address change, and flag that we need to check for conflicting Nodes
+	if oldIpv4 == nil || !node.Spec.BGP.IPv4Address.IP.Equal(oldIpv4.IP) {
+		checkConflicts = true
```

Comment:
nit:  Do you really need to define checkConflicts?  You could just return true here.